src/function/getTypeOf.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Returns the type of the value.
+ *
+ * @param {unknown} value - The value whose type needs to be determined.
+ * @return {string} The type of the value.
+ *
+ * @example
+ *
+ * getTypeOf([])            // 'array'
+ * getTypeOf({})            // 'object'
+ * getTypeOf(null)          // 'null'
+ * getTypeOf(new Map())     // 'map'
+ * getTypeOf(new Date())    // 'date'
+ * getTypeOf(new Error())   // 'error'
+ */
+export function getTypeOf(value: unknown): string {
```

Comment:
It's better to return `TypeName` instead of `string` to have autocomplete.
And of course, we should have a runtime check that the output is following the possible `TypeName`s